Source/Core/DolphinTool/ConvertCommand.cpp

```diff
@@ -0,0 +1,281 @@
+
+  std::optional<int> compression_level_o;
+  if (options.is_set("compression_level"))
+    compression_level_o = static_cast<int>(options.get("compression_level"));
+
+  if (format == DiscIO::BlobType::WIA || format == DiscIO::BlobType::RVZ)
+  {
+    if (!compression_o.has_value())
+    {
+      std::cerr << "Error: Compression format must be set for WIA or RVZ" << std::endl;
+      return 1;
+    }
+
+    if (compression_o.value() == DiscIO::WIARVZCompressionType::Zstd &&
+        format == DiscIO::BlobType::WIA)
+    {
+      std::cerr << "Error: WIA does not support Zstd" << std::endl;
+      return 1;
+    }
+
+    if (compression_o.value() == DiscIO::WIARVZCompressionType::None)
+    {
+      compression_level_o = 0;
+    }
+    else
+    {
+      if (!compression_level_o.has_value())
+      {
+        std::cerr << "Error: Compression level must be set when compression type is not 'none'"
+                  << std::endl;
+        return 1;
+      }
+
+      const std::pair<int, int> range = DiscIO::GetAllowedCompressionLevels(compression_o.value());
+      if (compression_level_o.value() < range.first || compression_level_o.value() > range.second)
+      {
+        std::cerr << "Error: Compression level not in acceptable range" << std::endl;
+        return 1;
+      }
+    }
+  }
+
+  // Open the blob reader
+  std::unique_ptr<DiscIO::BlobReader> blob_reader =
+      scrub ? DiscIO::ScrubbedBlob::Create(input_file_path) :
+              DiscIO::CreateBlobReader(input_file_path);
+  if (!blob_reader)
+  {
+    std::cerr << "Error: Unable to process disc image. If --scrub is enabled, try again without it."
+              << std::endl;
+    return 1;
+  }
+
+  // Perform the conversion
+  const auto NOOP_STATUS_CALLBACK = [](const std::string& text, float percent) { return true; };
+
+  bool success = false;
+
+  switch (format)
+  {
+  case DiscIO::BlobType::PLAIN:
+    success = DiscIO::ConvertToPlain(blob_reader.get(), input_file_path, output_file_path,
+                                     NOOP_STATUS_CALLBACK);
+    break;
+
+  case DiscIO::BlobType::GCZ:
+    success = DiscIO::ConvertToGCZ(blob_reader.get(), input_file_path, output_file_path,
+                                   volume->GetVolumeType() == DiscIO::Platform::WiiDisc ? 1 : 0,
+                                   block_size_o.value(), NOOP_STATUS_CALLBACK);
+    break;
+
+  case DiscIO::BlobType::WIA:
+  case DiscIO::BlobType::RVZ:
+    success = DiscIO::ConvertToWIAOrRVZ(blob_reader.get(), input_file_path, output_file_path,
+                                        format == DiscIO::BlobType::RVZ, compression_o.value(),
+                                        compression_level_o.value(), block_size_o.value(),
+                                        NOOP_STATUS_CALLBACK);
+    break;
+
+  default:
+    ASSERT(false);
+    break;
+  }
+
+  if (success)
+  {
+    std::cout << "Success" << std::endl;
```

Comment:
It's customary for CLI tools to not print anything on success, so unless a successful run results in any messages being printed other than this one, perhaps we should remove this message?